tests/CopyFilesTaskIntegrationTest.php

```diff
@@ -13,85 +14,106 @@ class CopyFilesTaskIntegrationTest extends \PHPUnit\Framework\TestCase
+			'logMessagePriority' => Project::MSG_INFO,
+		];
+
+		yield 'copy file with absolute path' => [
+			'target' => 'testCopyFileWithAbsolutePath',
+			'sourceFileName' => '/foo',
+			'targetFileName' => '/foo-copy',
+			'existingSourceFileContents' => 'FOO',
+			'existingTargetFileContents' => null,
+			'expectedTargetFileContents' => 'FOO',
+			'logMessageRegExp' => '~Copying.+/foo.+->.+/foo-copy~',
+			'logMessagePriority' => Project::MSG_INFO,
+		];
+
+		yield 'target file exists' => [
+			'target' => 'testTargetFileExists',
+			'sourceFileName' => '/new',
+			'targetFileName' => '/existing',
+			'existingSourceFileContents' => 'NEW',
+			'existingTargetFileContents' => 'EXISTING',
+			'expectedTargetFileContents' => 'EXISTING',
+			'logMessageRegExp' => '~/existing.+already exists.+SKIPPING~',
+			'logMessagePriority' => Project::MSG_INFO,
+		];
+
+		yield 'target file exists skip' => [
+			'target' => 'testTargetFileExistsSkip',
+			'sourceFileName' => '/new',
+			'targetFileName' => '/existing',
+			'existingSourceFileContents' => 'NEW',
+			'existingTargetFileContents' => 'EXISTING',
+			'expectedTargetFileContents' => 'EXISTING',
+			'logMessageRegExp' => '~/existing.+already exists.+SKIPPING~',
+			'logMessagePriority' => Project::MSG_INFO,
+		];
+
+		yield 'target file exists replace' => [
+			'target' => 'testTargetFileExistsReplace',
+			'sourceFileName' => '/new',
+			'targetFileName' => '/existing',
+			'existingSourceFileContents' => 'NEW',
+			'existingTargetFileContents' => 'EXISTING',
+			'expectedTargetFileContents' => 'NEW',
+			'logMessageRegExp' => '~/existing.+already exists~',
+			'logMessagePriority' => Project::MSG_VERBOSE,
+		];
 	}
 
-	public function testCopyFileWithAbsolutePath(): void
+	/**
+	 * @dataProvider copyFileDataProvider
+	 *
+	 * @param string $target
+	 * @param string $sourceFileName
+	 * @param string $targetFileName
+	 * @param string $existingSourceFileContents
+	 * @param string|null $existingTargetFileContents
+	 * @param string $expectedTargetFileContents
+	 * @param string $logMessageRegExp
+	 * @param int $logMessagePriority
+	 */
+	public function testCopyFile(
+		string $target,
+		string $sourceFileName,
+		string $targetFileName,
+		string $existingSourceFileContents,
+		?string $existingTargetFileContents,
+		string $expectedTargetFileContents,
+		string $logMessageRegExp,
+		int $logMessagePriority
+	): void
 	{
-		$sourceFilePath = self::TEMP_DIRECTORY_PATH . '/foo';
-		file_put_contents($sourceFilePath, 'FOO');
-		$targetFilePath = self::TEMP_DIRECTORY_PATH . '/foo-copy';
-		if (file_exists($targetFilePath)) {
+		$sourceFilePath = self::TEMP_DIRECTORY_PATH . $sourceFileName;
+		file_put_contents($sourceFilePath, $existingSourceFileContents);
+
+		$targetFilePath = self::TEMP_DIRECTORY_PATH . $targetFileName;
+		if ($existingTargetFileContents !== null) {
+			file_put_contents($targetFilePath, $existingTargetFileContents);
+		} else {
 			unlink($targetFilePath);
```

Comment:
File must exist before it is deleted, so the condition must be retained as it was before.

Reply:
The condition is back, sorry for that.